src/lib.rs

```diff
@@ -377,10 +382,18 @@ where
             uri.parse().into_future()
         };
         let instance = self.clone();
+        let uri2 = uri.clone();
         let body2 = body.clone();
         let method2 = method.clone();
         let response = url.map_err(Error::from).and_then(move |url| {
             let mut req = Request::builder();
+
+            if method2 == Method::GET {
+                if let Ok(etag) = lookup_etag(&uri2) {
```

Comment:
I like what you're going for here. I'm wondering if you had thoughts on making this an opt in feature and if what those opt in interfaces might look like. I might consider this suprising behavior in environments that restrict where you can write to disk and if you can write to disk.